From 759db416c81301833740a8f43ec3ffb738f1ce4c Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Sun, 9 Jun 2024 12:50:30 -0500
Subject: [PATCH] wined3d: Feed WINED3D_RS_POINTSIZE through a push constant
 buffer.

Not yet used by the GLSL backend, but will be needed by the HLSL backend.
---
 dlls/wined3d/stateblock.c      | 3 +++
 dlls/wined3d/wined3d_private.h | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/dlls/wined3d/stateblock.c b/dlls/wined3d/stateblock.c
index 80c1a6ea5b7..c064d950ba5 100644
--- a/dlls/wined3d/stateblock.c
+++ b/dlls/wined3d/stateblock.c
@@ -1616,6 +1616,7 @@ void CDECL wined3d_stateblock_set_render_state(struct wined3d_stateblock *stateb
                 stateblock->changed.alpha_to_coverage = 1;
                 stateblock->stateblock_state.alpha_to_coverage = (value == WINED3D_ALPHA_TO_COVERAGE_ENABLE);
             }
+            stateblock->changed.point_scale = 1;
             break;
 
         case WINED3D_RS_TEXTUREFACTOR:
@@ -3810,6 +3811,8 @@ void CDECL wined3d_device_apply_stateblock(struct wined3d_device *device,
     {
         struct wined3d_ffp_point_constants constants;
 
+        constants.size = int_to_float(state->rs[WINED3D_RS_POINTSIZE]);
+
         if (state->rs[WINED3D_RS_POINTSCALEENABLE])
         {
             float scale_factor = state->viewport.height * state->viewport.height;
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index 39a4aa5da9c..3e797b47795 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -2784,7 +2784,7 @@ struct wined3d_ffp_vs_constants
     struct wined3d_ffp_point_constants
     {
         float scale_const, scale_linear, scale_quad;
-        float padding; /* For the HLSL backend. */
+        float size;
     } point;
     struct wined3d_material material;
     float padding[3]; /* For the HLSL backend. */
-- 
GitLab

